### ISSUE TITLE
Document config values for contact form

### ISSUE THREAD

**Issue by mjgiarlo** [samvera/hyrax#113]
<a href="https://github.com/hackmastera"><img src="https://avatars.githubusercontent.com/u/845363?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [hackmastera](https://github.com/hackmastera)**
_Thursday Feb 18, 2016 at 19:33 GMT_
_Originally opened as https://github.com/projecthydra/sufia/issues/1587_

----

### Descriptive summary

Versions: all
### Expected behavior

Should be easy to figure out how to configure stuff
### Actual behavior

Have to dig and troubleshoot to find these values
### Related work

Add these config values to readme and/or config file generators:

https://github.com/projecthydra/sufia/blob/fc2dc3d86afcd3f53919b301ed036a648e260190/app/models/contact_form.rb#L24-L25

**Comment by mjgiarlo** [comment]
<a href="https://github.com/mjgiarlo"><img src="https://avatars.githubusercontent.com/u/131982?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mjgiarlo](https://github.com/mjgiarlo)**
_Thursday Feb 18, 2016 at 20:25 GMT_

----

I believe those values are cruft from the original port of ScholarSphere to Sufia, and are not used by Sufia. It'd be good to hear from the PSU folks (@awead, @cam156, @hectorcorrea) about whether this feature is worth porting from ScholarSphere into Sufia. If not, we should excise those two lines.

Thanks for finding the cruft, @hackmastera!

**Comment by mjgiarlo** [comment]
<a href="https://github.com/awead"><img src="https://avatars.githubusercontent.com/u/312085?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [awead](https://github.com/awead)**
_Thursday Feb 18, 2016 at 20:53 GMT_

----

Looks like we don't use that. We have a separate form.

**Comment by mjgiarlo** [comment]
<a href="https://github.com/mjgiarlo"><img src="https://avatars.githubusercontent.com/u/131982?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mjgiarlo](https://github.com/mjgiarlo)**
_Thursday Feb 18, 2016 at 21:01 GMT_

----

~~:+1: Cruft. Let's remove it.~~

**see below**

**Comment by mjgiarlo** [comment]
<a href="https://github.com/hackmastera"><img src="https://avatars.githubusercontent.com/u/845363?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [hackmastera](https://github.com/hackmastera)**
_Thursday Feb 18, 2016 at 21:03 GMT_

----

Okay I thought those were for setting up the contact form. Haven't been able to try it due to internet issues here this afternoon. If it's not the way to configure default contact form I remain utterly lost on that front.

**Comment by mjgiarlo** [comment]
<a href="https://github.com/mjgiarlo"><img src="https://avatars.githubusercontent.com/u/131982?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mjgiarlo](https://github.com/mjgiarlo)**
_Thursday Feb 18, 2016 at 21:54 GMT_

----

OK, I think I disagree with my earlier self that this is cruft, now that @hackmastera inspired me to dig deeper. I share the following as perhaps interesting "discovery" information.

Here's what the GW folks (@kercher, @kilahimm) did to get their contact form working and wired up:

https://github.com/gwu-libraries/gw-sufia#configure-contact-form-emailing

@awead -- after looking into this more, I think you Penn Staters _do_ use the Sufia form, though you have the config in your env-specific Rails configs so that the contact form values differ per environment (though the `Sufia::Engine.configure` is most definitely old-style -- now we use `Sufia.config` instead):

https://github.com/psu-stewardship/scholarsphere/blob/5eea37b92388fbebc9a1339f184cc6142ac52e1f/config/environments/development.rb#L3-L4 

The Albertans wanted to customize their contact form, so they created their own form based on what Sufia already has (probably via copypasta, but IMO it may be premature to consider modulizing the contact form for easier extension/customization):

https://github.com/ualbertalib/HydraNorth/blob/adea71d5ba7e4a29125daf5d5ea68cb056815180/config/environments/development.rb#L17-L23

So Sufia **does** use the undocumented `contact_email` or `from_email` config values; we should document these better and add them to the Sufia initializer, perhaps with more documentation recommending that folks might want to break these values out into `config/environments/*.rb` per Sufia best practice (as evidenced by PSU, GW, and UA)?

Thoughts on this approach? Was it even clear? @awead @hackmastera @jcoyne

**Comment by mjgiarlo** [comment]
<a href="https://github.com/mjgiarlo"><img src="https://avatars.githubusercontent.com/u/131982?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mjgiarlo](https://github.com/mjgiarlo)**
_Thursday Feb 18, 2016 at 22:04 GMT_

----

Here's the customization that Penn State added to the ScholarSphere contact form:

https://github.com/psu-stewardship/scholarsphere/blob/develop/app/controllers/contact_form_controller.rb

Basically, they wanted to inject some extra behavior after the contact form was submitted to let the user know what to expect of the submission. And they store their configs for that here:

https://github.com/psu-stewardship/scholarsphere/blob/5eea37b92388fbebc9a1339f184cc6142ac52e1f/config/initializers/sufia.rb#L11-L23

**Comment by mjgiarlo** [comment]
`config_email` is now in the Hyrax initializer and `from_email` is no longer used. Closing.
